x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/actions_log_filter.tsx

```diff
@@ -25,19 +25,79 @@ export const ActionsLogFilter = memo(
     onChangeFilterOptions: (selectedOptions: string[]) => void;
   }) => {
     const getTestId = useTestIdGenerator('response-actions-list');
+
+    // popover states and handlers
+    const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+    const onPopoverButtonClick = useCallback(() => {
+      setIsPopoverOpen(!isPopoverOpen);
+    }, [setIsPopoverOpen, isPopoverOpen]);
+    const onClosePopover = useCallback(() => {
+      setIsPopoverOpen(false);
+    }, [setIsPopoverOpen]);
+
+    // search string state
+    const [searchString, setSearchString] = useState('');
     const {
+      areHostsSelectedOnMount,
+      isLoading,
       items,
       setItems,
       hasActiveFilters,
       numActiveFilters,
       numFilters,
+      setAreHostsSelectedOnMount,
       setUrlActionsFilters,
+      setUrlHostsFilters,
       setUrlStatusesFilters,
-    } = useActionsLogFilter(filterName, isFlyout);
+    } = useActionsLogFilter({
+      filterName,
+      isFlyout,
+      isPopoverOpen,
+      searchString,
+    });
+
+    // track popover state to pin selected options
+    const wasPopoverOpen = useRef(isPopoverOpen);
+    useEffect(() => {
+      return () => {
+        wasPopoverOpen.current = isPopoverOpen;
+      };
+    }, [isPopoverOpen, wasPopoverOpen]);
+
+    // compute if selected hosts should be pinned
+    const shouldPinSelectedHosts = useCallback(
+      (isNotChangingOptions: boolean = true) => {
+        // case 1: when no hosts are selected initially
+        return (
+          isNotChangingOptions && wasPopoverOpen.current && isPopoverOpen && filterName === 'hosts'
+        );
+      },
+      [filterName, isPopoverOpen, wasPopoverOpen]
+    );
+
+    // augmented options based on hosts filter
+    const sortedHostsFilterOptions = useMemo(() => {
+      if (shouldPinSelectedHosts() || areHostsSelectedOnMount) {
+        const sortedItems = items.reduce<{ selected: FilterItems; rest: FilterItems }>(
```

Comment:
Could you use `orderBy` from lodash instead? You sort by `check` and order `desc`.

Reply:
Yeah, that would make this easier to read.

Reply:
e5b6447e14ede9b7867d29a3e26bd207dc27fad6